pynws/forecast.py

```diff
@@ -19,21 +30,21 @@
     r")?$"
 )
 
-ONE_HOUR = timedelta(hours=1)
+ONE_HOUR: Final[timedelta] = timedelta(hours=1)
 
 DetailValue = Union[int, float, list, str, None]
-_TimeValues = List[Tuple[datetime, datetime, DetailValue]]
+_TimeValue = Tuple[datetime, datetime, DetailValue]
 
 
 class DetailedForecast:
     """Class to retrieve forecast values for a point in time."""
 
-    def __init__(self: DetailedForecast, properties: Dict[str, Any]):
+    def __init__(self: DetailedForecast, properties: Mapping[str, Any]):
         if not isinstance(properties, dict):
             raise TypeError(f"{properties!r} is not a dictionary")
 
         self.update_time = datetime.fromisoformat(properties["updateTime"])
-        self.details: Dict[Detail, _TimeValues] = {}
+        self.details: Mapping[Detail, Sequence[_TimeValue]] = {}
```

Comment:
I think we should have all input parameters as more generic types, like you have done for `properties`, but anything that is definitely a `dict` like `self.details` should be explicitly typed as `Dict`.  Like this comment https://stackoverflow.com/questions/52487663/python-type-hints-typing-mapping-vs-typing-dict#comment99114819_52487800

Reply:
I reverted to using `Dict` and `List`. The `properties` case you mentioned is a good example, but the code currently is currently ensuring that the value is a `dict`, so accepting a `Mapping` is a lie.